Added Login & Signup UI #10

asadsid95 opened the pull request:
This is what I've got so far for the front-end. I am far from being done but I wanted to share what I have so far and would appreciate any feedback (YES, I'm expecting to be torn to pieces but I really that).

For **/login & /register:**

I haven't been able to figure out the following items related to the front-end:

- how to change the font for the header
- theme colours of the textfields (I.e. when I click on it and the pre-selected theme is blue-ish) and the checkbox
- size of the submit button
- how to show ‘forgot password’ inside of a textfield (so thats why I made another ‘FormControlLabel’ temporarily)
- changing colour of the divider
- HelpText, snackbars

I sent up very basic routes from /Landingpage (created it for the sake of simplicity); I got it to render path-specific components but I know that I may be missing a lot more but not exaclty sure what I'm missing so feedback would be really helpful!

For Material-UI, I went about the basic route so that I can do the basic styling but creating custom themes is something I'm still not understanding

I also haven't been able to figure out what to do once user info and new member info is provided; Not sure if thats another ticket that I'm assigned but would love any pointers on that :)

asadsid95 committed:
Issue #10

This is what I've got so far for the front-end. I am far from being done but I wanted to share what I have so far and would appreciate any feedback.
For /login & /register:

I haven't been able to figure out the following items related to the front-end:

- how to change the font for the header
- theme colours of the textfields (I.e. when I click on it and the pre-selected theme is blue-ish) and the checkbox
- size of the submit button
- how to show ‘forgot password’ inside of a textfield (so thats why I made another ‘FormControlLabel’ temporarily)
- changing colour of the divider
- HelpText, snackbars

I sent up very basic routes from Landing page (created it for the sake of simplicity); I got it to render path-specific components but I know that I may be missing a lot more but not exaclty sure what I'm missing so feedback would be really helpful!

For Material-UI, I went about the basic route so that I can do the basic styling but creating custom themes is something I'm still not understanding

asadsid95 committed:
Changed location of /login, /register, /Landingpage

asadsid95 committed:
comment for testing

aeciorc commented on client/src/index.js:
```diff
@@ -1,10 +1,17 @@
+//import App from './App';
 import * as serviceWorker from './serviceWorker';
+//import Signup from './Signup'
+import Login from './pages/login'
```
Some of these imports are not used, so let's remove them

aeciorc commented on client/src/index.js:
```diff
@@ -1,10 +1,17 @@
-ReactDOM.render(<App />, document.getElementById('root'));
+ReactDOM.render(<Router>
+    <App />
+    </Router>, document.getElementById('root'));
```
I think it's best to add the router in the App component

aeciorc commented on client/src/pages/login.js:
```diff
@@ -0,0 +1,89 @@
+        display: 'flex',
+        flexDirection: 'column',
+        alignItems: 'center',
+        fontFamily: '"ProximaNova"', //Not sure how to change font
```
If it's not one of the default fonts, you need to add it first to make it available. Easiest way is to use a cdn and add it to public/index.html, for example:
<link href="https://fonts.googleapis.com/css?family=Titillium+Web:300,400,700" rel="stylesheet">

Look for a CDN for proximaNova, if you can't find it, follow the self-hosting tutorial here:
https://scotch.io/@micwanyoike/how-to-add-fonts-to-a-react-project

aeciorc replied:
Also, if you want to use that font in every typography component, you would do that on the theme:
```
typography: {
    fontFamily: "ProximaNova"
  },
```

aeciorc commented on client/src/pages/login.js:
```diff
@@ -0,0 +1,89 @@
+        
+        <div className={classes.containerTweaks}>
+            <div className={classes.pageInfoTweaks}>
+                <Typography component="h1" variant="h4">Member login</Typography>
```
remove the component prop here, if you need h1 do that in the variant

aeciorc commented on client/src/themes/theme.js:
```diff
@@ -9,3 +10,17 @@ export const theme = createMuiTheme({
+*/
+
+const theme = createMuiTheme({
+  palette: {
```
Should be:
```
palette: {
     main:{
    primary: "#69E781"
}
  },
```

aeciorc reviewed (commented):
When creating PRs and branches, use more descriptive names like "Added login UI #10" and "login-ui-10"

asadsid95 committed:
Added Login & Signup UI #10

Removed unnecessary/unused imports from prior commit

In /register.js, change type for confirm password field to 'password' to prevent plain text render

Removed unnecessary component declaration

Removed BrowserRouter from /index.js file
